app/src/main/java/com/example/cocktailmaster/ui/home/HomeFragment.kt

```diff
@@ -1,29 +1,51 @@
 package com.example.cocktailmaster.ui.home
 
 import android.view.View
+import androidx.appcompat.widget.SearchView
+import com.example.cocktailmaster.R
 import com.example.cocktailmaster.base.BaseFragment
 import com.example.cocktailmaster.data.model.Drink
 import com.example.cocktailmaster.databinding.FragmentHomeBinding
 import com.example.cocktailmaster.ui.RepositoryUtils
 import com.example.cocktailmaster.ui.home.adapter.AlphabetAdapter
+import com.example.cocktailmaster.ui.home.adapter.DrinksSearchAdapter
 import com.example.cocktailmaster.ui.home.adapter.RandomDrinksAdapter
 
-class HomeFragment : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::inflate),
-    HomeContract.View {
+class HomeFragment :
+    BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::inflate),
+    HomeContract.View,
+    View.OnClickListener,
+    SearchView.OnCloseListener,
+    SearchView.OnQueryTextListener {
 
     private var presenter: HomePresenter? = null
     private val drinkAdapter = RandomDrinksAdapter()
     private val alphabetAdapter = AlphabetAdapter()
+    private val drinkSearchAdapter = DrinksSearchAdapter()
     private val drinks = mutableListOf<Drink>()
 
     override fun initViews() {
-        binding.recyclerDrinks.adapter = drinkAdapter
-        binding.includeAlphabets.recyclerAlphabets.adapter = alphabetAdapter
+        listOf(binding.imageSearch, binding.imageFavourite).forEach {
+            it.setOnClickListener(this)
+        }
+        binding.apply {
+            recyclerDrinks.adapter = drinkAdapter
+            includeAlphabets.recyclerAlphabets.adapter = alphabetAdapter
```

Comment:
```suggestion
            recyclerAlphabets.adapter = alphabetAdapter
```
hình như viết v vẫn đk mà nhỉ